tests/conftest.py

```diff
@@ -52,26 +58,51 @@ def create_client(self):
         command = ['xterm', '-hold', '-class', wmclass, '-e', 'true']
         # enforce a hook when the window appears
         self.call('rule', 'once', 'class='+wmclass, 'hook=here_is_'+wmclass)
-        # run process that wait for the hook
+        # the following is still racy because xterm might connect to X before
+        # the herbstclient --wait does.
+        # 1. run process that wait for the hook
         hc_wait = subprocess.Popen([self.HC_PATH, '--wait', 'rule', 'here_is_'+wmclass],
                     env=self.env,
                     stdout=subprocess.PIPE)
-        # start the process...
+        # 2. start the process...
         proc = subprocess.Popen(command,env=self.env)
         # once the window appears, the hook is fired, and the --wait exits:
-        hc_wait.wait(2)
+        try:
+            if hc_wait.wait(2) != 0:
+                self.hlwm_process.client_timeout( \
+                    'waiting for client triggering the hook {}'.format(wmclass))
+        except subprocess.TimeoutExpired:
+            self.hlwm_process.client_timeout( \
+                'waiting for client triggering the hook {}'.format(wmclass))
         winid = hc_wait.stdout.read().decode().rstrip('\n').split('\t')[-1]
 
         return SimpleNamespace(proc=proc, winid=winid)
 
 
 @pytest.fixture
-def hlwm():
+def hlwm(hlwm_process):
     display = os.environ['DISPLAY']
     #display = ':13'
-    return HlwmBridge(display)
-
-
+    return HlwmBridge(display,hlwm_process)
+
+
+class HlwmProcess:
+    def __init__(self,proc):
+        self.proc = proc
+
+    def client_timeout(self,reason):
+        """if a client complains about a timeout, investigate the
```

Comment:
By "client" you're specifically referring to herbstclient, right? Either way, a clarification would probably be good, given the potential for confusion within this file.